frontend/public/components/storage/volume-modes/volume-mode.tsx

```diff
@@ -0,0 +1,99 @@
+import * as React from 'react';
+
+import { RadioInput } from '@console/internal/components/radio';
+import { FormGroup } from '@patternfly/react-core';
+import { PersistentVolumeClaimKind } from '@console/internal/module/k8s';
+import { FieldLevelHelp } from '@console/internal/components/utils';
+
+import { volumeModeRadios, getVolumeModeForProvisioner, initialVolumeModes } from '../shared';
+
+export const VolumeModeSelector: React.FC<VolumeModeSelectorProps> = (props) => {
+  const {
+    formClassName,
+    pvcResource,
+    accessMode,
+    onChange,
+    provisioner,
+    storageClass,
+    availableVolumeMode,
+    loaded,
+    label,
+  } = props;
+
+  const pvcInitialVolumeMode: string = pvcResource
+    ? pvcResource?.spec?.volumeMode
+    : availableVolumeMode ?? initialVolumeModes[0];
+
+  const [volumeMode, setVolumeMode] = React.useState('');
+  const [allowedVolumeModes, setAllowedVolumeModes] = React.useState<string[]>([]);
+
+  React.useEffect(() => {
+    if (loaded) {
+      setAllowedVolumeModes(getVolumeModeForProvisioner(provisioner, accessMode));
+    }
+  }, [loaded, provisioner, accessMode]);
+
+  React.useEffect(() => {
+    // Makesure the default or already checked radio button value is from any one of allowed the access mode
+    if (allowedVolumeModes.length > 0) {
+      if (volumeMode === '' && allowedVolumeModes.includes(pvcInitialVolumeMode)) {
```

Comment:
Unless you really need to distinguish between `''` and `null` or `undefined` use

```suggestion
      if (!volumeMode && allowedVolumeModes.includes(pvcInitialVolumeMode)) {
```